dotCMS/src/main/java/com/dotcms/rest/api/v1/storage/ContentStorageResource.java

```diff
@@ -0,0 +1,218 @@
+import javax.ws.rs.Path;
+import javax.ws.rs.Produces;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Context;
+import javax.ws.rs.core.MediaType;
+import javax.ws.rs.core.Response;
+import javax.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.util.List;
+import java.util.Map;
+import java.util.zip.ZipEntry;
+import java.util.zip.ZipOutputStream;
+
+/**
+ * Resource to expose the storage and metadata
+ * @author jsanca
+ */
+@Path("/v1/storage")
+public class ContentStorageResource {
+
+    private final WebResource webResource;
+    private final FileStorageAPI fileStorageAPI;
+    private final MultiPartUtils multiPartUtils;
+    private final ContentStorageHelper helper;
+    private final ContentletMetadataAPI contentletMetadataAPI;
+    private final LanguageWebAPI languageWebAPI;
+
+    @VisibleForTesting
+    ContentStorageResource(final FileStorageAPI fileStorageAPI,
+            final WebResource webResource,
+            final MultiPartUtils multiPartUtils,
+            final ContentStorageHelper helper,
+            final ContentletMetadataAPI contentletMetadataAPI, final LanguageWebAPI languageWebAPI) {
+
+        this.fileStorageAPI = fileStorageAPI;
+        this.webResource    = webResource;
+        this.multiPartUtils = multiPartUtils;
+        this.helper = helper;
+        this.contentletMetadataAPI = contentletMetadataAPI;
+        this.languageWebAPI = languageWebAPI;
+    }
+
+    public ContentStorageResource() {
+        this(APILocator.getFileStorageAPI(), new WebResource(),
+                new MultiPartUtils(), new ContentStorageHelper(),
+                APILocator.getContentletMetadataAPI(), WebAPILocator.getLanguageWebAPI());
+    }
+
+    @GET
+    @JSONP
+    @NoCache
+    @Produces(MediaType.APPLICATION_OCTET_STREAM)
+    public final Response getFileFromStorage(@Context final HttpServletRequest request,
+                                           @Context final HttpServletResponse response,
+                                           @QueryParam("path") final String path) {
+
+        final File   file        = StoragePersistenceProvider.INSTANCE.get().getStorage().pullFile("files", path);
+        final String fileName    = FilenameUtils.getName(path);
+
+        return Response.ok(
+                new StreamingOutput() {
+                    @Override
+                    public void write(OutputStream output)  {
+                        try (ZipOutputStream zoutput = new ZipOutputStream(output);
+                                InputStream fileInput = Files.newInputStream(file.toPath())) {
+
+                            int   bytesRead      = 0;
+                            final byte[] buffer  = new byte[512];
+                            zoutput.putNextEntry(new ZipEntry(path));
+
+                            while ((bytesRead = fileInput.read(buffer)) > 0) {
```

Comment:
Codacy found an issue: [Avoid assignments in operands](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=6220272)